### PR TITLE
[main] Upgrade Rust toolchain

### PR DESCRIPTION
**Pull request by onalante-msft**
This is our monthly toolchain upgrade to the latest stable version. Since iotedge/edgelet is a binary package, we want to be tracking Rust compiler releases closely in the event a stdlib vulnerability is found. We do not pin to "stable", however, since that has caused pipeline breakage when some code patterns are made illegal (like in the upgrade from 1.47 to 1.48 [^0]) or, more often, clippy lints are added.

- Add `#[must_use]` directives on methods returning `Self`
- Change `unwrap_or_else(...::new)` to `unwrap_or_default` where appropriate
- Box `CertIssuanceOptions` in `EdgeCa` enum due to large variant size difference otherwise

## Azure IoT Edge PR checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines and Best Practices
- [x] I have read the [contribution guidelines](https://github.com/azure/iotedge#contributing).
- [x] Title of the pull request is clear and informative.
- [x] Description of the pull request includes a concise summary of the enhancement or bug fix.

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.
- Description of the pull request includes 
	- [ ] concise summary of tests added/modified
	- [ ] local testing done.

[^0]: https://github.com/rust-lang/rust/blob/master/RELEASES.md#compatibility-notes-11
  Namely, the point on `mem::uninitialized`.

### CODE REVIEW

File: rust-toolchain.toml [hunk 1]
```diff
@@ -1,2 +0,0 @@
-[toolchain]
```
Comment (arsing) [inline rust-toolchain.toml:1]: Why go back to the unsuffixed file?

Comment (onalante-msft) [reply rust-toolchain.toml:1]: Alright, that sounds good to me.

**Comment by arsing** [comment]
>- Add `#[must_use]` directives on methods returning `Self`

Just `#![allow]` it on the crate. It's on [the list](https://github.com/Azure/iot-identity-service/blob/8f53f29c063c03529e508ee4111c7f1b1110c08f/CONTRIBUTING.md?plain=1#L70) of "lints that are okay to allow on the whole crate because they suck at their job."

**Comment by onalante-msft** [comment]
It was firing on `with_...` functions for the most part, and I think it is reasonable to have `#[must_use]` on those.

**Review by arsing — APPROVED** [review]

**Review by gordonwang0 — APPROVED** [review]

**Pull request merged**